ApiBundle/Controller/AreaController.php

```diff
@@ -118,16 +136,198 @@ public function moveBlockFromAreaToAreaAction(Request $request, $nodeId)
     }
 
     /**
+     * @param Request $request
+     * @param string  $areaParentId
+     * @param string  $templateId
+     *
+     * @Config\Route("/{areaParentId}/{templateId}/move_area", name="open_orchestra_api_area_move_in_template")
+     * @Config\Method({"POST"})
+     *
+     * @Config\Security("is_granted('ROLE_UPDATE_TREE_TEMPLATE')")
+     *
+     * @return Response
+     */
+    public function moveAreaInTemplateAction(Request $request, $areaParentId, $templateId)
+    {
+        $areaFacade = $this->get('jms_serializer')->deserialize($request->getContent(), 'OpenOrchestra\ApiBundle\Facade\AreaFacade', $request->get('_format', 'json'));
+
+        $template = $this->get('open_orchestra_model.repository.template')->findOneByTemplateId($templateId);
+        $areaParent = $this->get('open_orchestra_model.repository.template')->findAreaInTemplateByAreaId($template, $areaParentId);
+
+        $this->get('open_orchestra_api.transformer.area')->reverseTransform($areaFacade, $areaParent);
+        $this->get('object_manager')->flush();
+
+        return array();
+    }
+
+    /**
+     * @param Request $request
+     * @param string  $areaParentId
+     * @param string  $nodeId
+     * @param string  $language
+     * @param string  $version
+     * @param string  $siteId
+     *
+     * @Config\Route("/{areaParentId}/{siteId}/{nodeId}/{version}/{language}/move_area", name="open_orchestra_api_area_move_in_node")
+     * @Config\Method({"POST"})
+     *
+     * @return Response
+     */
+    public function moveAreaInNodeAction(Request $request, $areaParentId, $nodeId, $language, $version, $siteId)
+    {
+        $areaFacade = $this->get('jms_serializer')->deserialize($request->getContent(), 'OpenOrchestra\ApiBundle\Facade\AreaFacade', $request->get('_format', 'json'));
+
+        $node = $this->get('open_orchestra_model.repository.node')->findVersion($nodeId, $language, $siteId, $version);
+        $this->denyAccessUnlessGranted($this->getEditionRole($node), $node);
+
+        $areaParent = $this->get('open_orchestra_model.repository.node')->findAreaInNodeByAreaId($node, $areaParentId);
+
+        $this->get('open_orchestra_api.transformer.area')->reverseTransform($areaFacade, $areaParent);
+        $this->get('object_manager')->flush();
+
+        return array();
+    }
+
+    /**
+     * @param string      $areaId
+     * @param string      $templateId
+     * @param string|null $areaParentId
+     *
+     * @Config\Route("/{areaId}/delete-column-in-template/{templateId}/{areaParentId}", name="open_orchestra_api_area_column_delete_in_template")
+     * @Config\Route("/{areaId}/delete-row-in-template/{templateId}", name="open_orchestra_api_area_row_delete_in_template")
+     * @Config\Method({"DELETE"})
+     *
+     * @Config\Security("is_granted('ROLE_ACCESS_UPDATE_GENERAL_NODE')")
+     *
+     * @return Response
+     */
+    public function deleteAreaTemplateAction($areaId, $templateId, $areaParentId = null)
```

Comment:
@alavieille : moveAreaInBlock and deleteAreaTemplate. It could be nice to homogenize the naming